Source/WebCore/style/StyleScope.cpp

```diff
@@ -791,20 +791,14 @@ auto Scope::collectResolverScopes() -> ResolverScopes
 {
     ASSERT(!m_shadowRoot);
 
-    if (!resolverIfExists())
-        return { };
-
     ResolverScopes resolverScopes;
-
-    resolverScopes.add(*resolverIfExists(), Vector<WeakPtr<Scope>> { this });
+    resolverScopes.add(resolver(), Vector<WeakPtr<Scope>> { this });
```

Comment:
I think the actual issue here is that we bail out if the top level resolver has been cleared but that does not mean there can't be live resolvers in shadow trees. Doing
```
    if (auto* resolver = resolverIfExists())
       resolverScopes.add(*resolver, Vector<WeakPtr<Scope>> { this });
```
seems to fix this.

Reply:
You should make a test case. The original bug says you need dual screens but I can repro just fine with a single screen. See fast/media/mq-inverted-colors-live-update.html, fast/media/mq-inverted-colors-live-update-in-subframes.htm, LayoutTests/accessibility/smart-invert.html for examples.

Reply:
To make a test for this you'll need to figure out how to delete the document style resolver without deleting the shadow tree ones. I think you can achieve this by using any non-dynamic media query (one without MediaQueryDynamicDependency in MediaQueryFeature.cpp) on a document stylesheet.

Reply:
I've changed it to your suggestion and added a test case.